activerecord/lib/active_record/relation.rb

```diff
@@ -346,7 +346,7 @@ def update_all(updates)
     #
     #   Person.where(age: 0..18).destroy_all
     def destroy_all
-      records.each(&:destroy).tap { reset }
+      records.map(&:destroy).select{ |record| record.present? }.tap { reset }
```

Comment:
Don't we want `.select(&:destroyed?)`?

Reply:
This can actually be shortened to just `select(&:destroy)`